### ISSUE TITLE
CLI stack validate outputs stack create API request instead of interpolated stack YAML

### ISSUE THREAD

**Issue by SpComb** [kontena/kontena#2887]
Looks like the spec is trying to match the wrong syntax for `links: ...`, it expects a list of plain of link names, not the `name/alias` objects:

```
stack validate
  service_link
 [done] Creating stack simple      
> Select link simple/redis
---
# /kontena/test/spec/fixtures/stack/service_link/kontena.yml
stack: test/test
version: 0.1.0
name: test
registry: file://
expose: 
services:
- instances: 
  image: nginx:1-alpine
  env: 
  links:
  - name: simple/redis
    alias: simple/redis
  external_links: []
  ports: []
  volumes: []
  volumes_from: []
  affinity: []
  stateful: false
  deploy_opts:
    wait_for_port: 
    min_health: 
    interval: 
  hooks: {}
  health_check:
    port: 
    protocol: 
    uri: 
    timeout: 
    interval: 
    initial_delay: 
  read_only: false
  log_driver: 
  log_opts: 
  name: nginx
volumes: []
dependencies: 
variables:
  lb: simple/redis
parent_name: 
 [done] Removing stack simple      
    allows to select matching service (FAILED - 1)

Failures:

  1) stack validate service_link allows to select matching service
     Failure/Error: expect(k.out.match(/links\:\r\n\s+- simple\/redis/)).to be_truthy
     
       expected: truthy value
            got: nil
     # ./spec/features/stack/validate_spec.rb:18:in `block (4 levels) in <top (required)>'
     # ./spec/support/shell.rb:33:in `block in with_fixture_dir'
     # ./spec/support/shell.rb:32:in `chdir'
     # ./spec/support/shell.rb:32:in `with_fixture_dir'
     # ./spec/features/stack/validate_spec.rb:11:in `block (3 levels) in <top (required)>'

Finished in 3.69 seconds (files took 0.1726 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/features/stack/validate_spec.rb:7 # stack validate service_link allows to select matching service
```

**Comment by SpComb** [comment]
Looks like #2823 changed what YAML `kontena stack validate` actually outputs: it used to be the `reader.fully_interpolated_yaml`, but now it's the `stack`.

**Comment by SpComb** [comment]
@kke bug or feature?

**Comment by kke** [comment]
Not sure. I think there was a reason but it might not have been a good one.

**Comment by kke** [comment]
Not sure if there's any reason to output anything except validation errors (and exit with error) and warnings.

**Comment by SpComb** [comment]
I think `kontena stack validate` could have been used as a tool to debug things like opto interpolations... in that sense, either the interpolated YAML or the server JSON are both interesting things to show, the current JSON also includes the `ServiceGenerator` things.

The interpolated YAML might be easier to read, though.

**Comment by kke** [comment]
I think the reason might have been that the reader messes with the services hash, especially when doing `extends`, which mutates `fully_interpolated_yaml` and shows services without "extends" or any of the stuff that came from the extended service.

**Comment by kke** [comment]
Maybe it should currently just dump JSON instead of YAML when `--json` given. Then it would be useful for someone trying to use/build their own API client.

**Comment by SpComb** [comment]
I think I'll count this "kontena stack validate outputs the generated API JSON (in YAML) instead of the interpolated stack YAML" as a known issue in the 1.4.0 changelogs... can we fix this to revert the changes in #2823 and have `kontena stack validate` output the interpolated stack YAML as before by default?

**Comment by SpComb** [comment]
Reclassified this from `chore` to `bug`
